src/core/border_router/routing_manager.cpp

```diff
@@ -683,6 +683,53 @@ bool RoutingManager::IsValidOnLinkPrefix(const Ip6::Prefix &aOnLinkPrefix)
            !aOnLinkPrefix.IsMulticast();
 }
 
+bool RoutingManager::IsExternalRoutePublished() const
+{
+    NetworkData::Iterator            iterator = NetworkData::kIteratorInit;
+    NetworkData::ExternalRouteConfig routeConfig;
+    bool                             isPublished = false;
+    uint16_t                         rloc16;
+
+    rloc16 = Get<Mle::MleRouter>().GetRloc16();
+
+    // Iterate through the external routes in the network data
+    while (Get<NetworkData::Leader>().GetNextExternalRoute(iterator, routeConfig) == kErrorNone)
+    {
+        // Check if the route is stable and matches the current RLOC16
+        if (routeConfig.mStable && (routeConfig.GetRloc16() == rloc16))
+        {
+            isPublished = true;
+            break;
+        }
+    }
+
+    return isPublished;
+}
+
+bool RoutingManager::IsDefaultRoutePublished() const
+{
+    NetworkData::Iterator            iterator = NetworkData::kIteratorInit;
+    NetworkData::ExternalRouteConfig routeConfig;
+    bool                             isPublished = false;
+    Ip6::Prefix                      defaultRoutePrefix = AsCoreType(&Ip6::kDefaultRoutePrefix);
+    uint16_t                         rloc16;
+
+    rloc16 = Get<Mle::MleRouter>().GetRloc16();
+
+    // Iterate through the external routes in the network data
+    while (Get<NetworkData::Leader>().GetNextExternalRoute(iterator, routeConfig) == kErrorNone)
+    {
+        // Check if the route is stable, matches the current RLOC16 and is the default route
+        if (routeConfig.mStable && (routeConfig.GetRloc16() == rloc16) && (routeConfig.GetPrefix() == defaultRoutePrefix))
+        {
+            isPublished = true;
+            break;
+        }
+    }
+
+    return isPublished;
+}
```

Comment:
We can retrieve the same information from the `NetworkData::Local` class, which tracks all network data entries added by this device, including entries pending registration.

Alternatively, if we want to get entries that are already registered with the leader, we can use the `NetworkData::Leader` class.

The `NetworkData` class (which is base class of `Local` and `Leader`) already provides related helper methods for this:

- `GetNextExternalRoute(Iterator &aIterator, uint16_t aRloc16, ExternalRouteConfig &aConfig)` allows filtering entries based on a given RLOC16.
`NetworkData::ContainsExternalRoute()` checks if a specific external route prefix entry is present in the network data (e.g., to check if this device has added a default route).

Reply:
+1. I think the current OT API should be sufficient to retrieve the desired data. Probably this method should be implemented in ot-br-posix, if it's for telemetry purpose.